### PR TITLE
T13558

### PR DESCRIPTION
**Pull request by carlocaione**
https://phabricator.endlessm.com/T13558

### CODE REVIEW

File: eos-tech-support/eos-write-live-image [hunk 1]
```diff
@@ -186,106 +237,139 @@ check_exists "$BOOT_ZIP.asc" "bootloader bundle signature"
...
+    xorriso -as mkisofs \
+            -graft-points \
+            -b endless/grub/i386-pc/iso/eltorito.img \
+            -no-emul-boot \
+            -boot-load-size 4 \
+            -boot-info-table \
+            --grub2-boot-info \
+            --grub2-mbr ${DIR_IMAGES_ENDLESS}/grub/i386-pc/iso/boot_hybrid.img \
+            --efi-boot endless/efi.img \
+            -efi-boot-part \
```
Comment (wjt) [inline eos-tech-support/eos-write-live-image]: What's going on with the weird combination of `-` and `--` prefixes on options? An idiosyncrasy of `xorriso`?

File: eos-tech-support/eos-write-live-image [hunk 2]
```diff
@@ -140,23 +177,37 @@ for command in "${!dependencies[@]}"; do
     fi
 done
 
-if [ ! -b "$DEVICE" ]; then
-    echo "$DEVICE does not exist or is not a block device... aborting!" >&2
-    exit 1
-fi
+if [ "$ISO" ]; then
+    if [ -f $OUTPUT ]; then
```
Comment (carlocaione) [inline eos-tech-support/eos-write-live-image]: agree

File: eos-tech-support/eos-write-live-image [hunk 3]
```diff
@@ -140,23 +177,37 @@ for command in "${!dependencies[@]}"; do
...
+        exit 1
+    fi
 
-if grep -qs "$DEVICE" /proc/mounts; then
-    # Protect against overwriting the device currently in use
-    echo "$DEVICE is currently in use -- please unmount and try again" >&2
-    exit 1
-fi
+    OUTPUT_DIR=$(dirname $OUTPUT)
+    mkdir -p $OUTPUT_DIR  &>/dev/null
```
Comment (carlocaione) [inline eos-tech-support/eos-write-live-image]: ok

File: eos-tech-support/eos-write-live-image (hunk 1 above)
Comment (carlocaione) [inline eos-tech-support/eos-write-live-image]: the `--` are passed down to `mkisofs`, nothing to worry about

**Pull request merged**